Makefile

```diff
@@ -96,10 +96,16 @@ depends = .doc/$(1) .install/$(1) .check/$(1) .test/$(1)
 
 ### Rules
 
-.PHONY: all install check test document shiny
+.PHONY: all install check test document shiny \
++               check_base check_models check_modules
 
 all: install document
 
+
+check_base: $(BASE_C) 
+check_models: $(MODELS_C) 
+check_modules: $(BASE_C) $(MODULES_C) 
```

Comment:
This seems to imply that modules depends on base, but models doesn't. Is this true? Doesn't make know the dependencies among packages? (e.g., if models already know which subset of base it needs to build, doesn't modules also). And these seem to be checks, not just builds -- running a check on all the base packages before running the modules checks seems redundant with the check_base

Reply:
hey @mdietze, I had a discussion with @robkooper on google meet few days back and The problem with modules is that, it has a circular dependency with base, which requires the base check to run first, also we have kept check_base and check_modules separately as we further plan to trigger the checks only when changes are made to those specific directories. So it would be wise for now to keep check_base and check_modules separate for this PR.

Reply:
Yes but doesn't the circular dependency have to do with running the *install* for base, not on running the *checks*. Just installing base would be quicker than also running the checks on all the base packages.

Reply:
Oh, yes you are right sir, I tried running the workflow with [installing the base packages and then running the check on modules](https://github.com/RohanSasne/pecan/actions/runs/5163977571) and it is surely quicker than before, I have made the necessary changes, please have a look. :)

Reply:
Cool, I have added comments to address why we install BASE first and then run check on modules.